scripts/globals/effects/magic_shield.lua

```diff
@@ -1,13 +1,23 @@
 -----------------------------------
+-- Magic Shield
+-- Blocks magic damage and effects depending on power
 --
--- Magic Shield BLOCKS all magic attacks
---
+-- Power Notes:
+--  0 - 50%  DMGMAGIC (e.g. Fool's Tonic)
```

Comment:
Just tossing you some info/possibilities here:

We do have additional field sin the effect we could be using, which could make this simpler.

`hasStatusEffect(effectID, subid)` is also a thing. one of the lines you replaced was using that to tell apart 2 versions of magic shield. We could also use subpower and tiers. Frankly what we're doing with power even before your PR should have been tiers but way back when somebody used power so we stuck with it. We could use subpower to decide if we're blocking status effects or not as an example instead of using an invalid dmg reduction of -101.